presto-postgresql/src/test/java/io/prestosql/plugin/postgresql/TestPostgreSqlTypeMapping.java

```diff
@@ -614,6 +655,39 @@ public void testTimestampWithTimeZone(boolean insertWithPresto)
         tests.execute(getQueryRunner(), dataSetup);
     }
 
+    @Test(dataProvider = "testTimestampWithTimeZoneDataProvider")
+    public void testArrayTimestampWithTimeZone(boolean insertWithPresto)
+    {
+        DataType<List<ZonedDateTime>> dataType;
+        DataSetup dataSetup;
+        if (insertWithPresto) {
+            dataType = arrayDataType(prestoTimestampWithTimeZoneDataType());
+            dataSetup = prestoCreateAsSelect("test_array_timestamp_with_time_zone");
+        }
+        else {
+            dataType = arrayDataType(postgreSqlTimestampWithTimeZoneDataType(), "timestamptz[]");
+            dataSetup = postgresCreateAndInsert("tpch.test_array_timestamp_with_time_zone");
+        }
+
+        DataTypeTest tests = DataTypeTest.create()
+                .addRoundTrip(dataType, asList(epoch.atZone(UTC), epoch.atZone(kathmandu)))
+                .addRoundTrip(dataType, asList(beforeEpoch.atZone(kathmandu), beforeEpoch.atZone(UTC)))
+                .addRoundTrip(dataType, asList(afterEpoch.atZone(UTC), afterEpoch.atZone(kathmandu)))
+                .addRoundTrip(dataType, asList(timeDoubledInJvmZone.atZone(UTC)));
+        if (!insertWithPresto) {
+            // Postgres results with non-DST time (winter time) for timeDoubledInJvmZone.atZone(jvmZone) while Presto results with DST time, so skip this test when insertWithPresto
```

Comment:
... and Presto behavior is consistent with Java Time:

```java
ZoneId bahia = ZoneId.of("America/Bahia_Banderas");
System.out.println(LocalDateTime.of(2018, 10, 28, 0, 33, 17, 456_000_000).atZone(bahia).toInstant());
System.out.println(LocalDateTime.of(2018, 10, 28, 1, 33, 17, 456_000_000).atZone(bahia).toInstant()); // timeDoubledInJvmZone
System.out.println(LocalDateTime.of(2018, 10, 28, 2, 33, 17, 456_000_000).atZone(bahia).toInstant());
```

prints

```
2018-10-28T05:33:17.456Z
2018-10-28T06:33:17.456Z // it's 1h after the previous one, still in summer time
2018-10-28T08:33:17.456Z
```

Did you try reporting this in Postgres?

Reply:
I think that the issue here is wider since there are really 2 different times between 01:00 and 02:00  and the expression of `2018-10-28 01:33:17 America/Bahia_Banderas` doesn't uniquely identify them. It can be either the time at summer time (DST on) or the time at winter time (DST off) after the clock move 1 hour back.
To my opinion both are right in this case.

Reply:
We should be inserting with _a point in time_ semantics, so we should get the same point in time back.
Thus, this should work. Please add a TODO here.